dt-backend/test/model/user/user.e2e-spec.ts

```diff
@@ -2,22 +2,47 @@ import { Test, TestingModule } from '@nestjs/testing';
 import { UserModule } from '../../../src/model/user/user.module';
 import * as request from 'supertest';
 import { INestApplication } from '@nestjs/common';
-import { AppModule } from '../../../src/app.module';
-import { UserController } from '../../../src/model/user/user.controller';
-import { UserService } from '../../../src/model/user/user.service';
+import { ModuleMocker } from 'jest-mock';
+import { User } from '../../../src/model/types/user.type';
 
 describe('UserController e2e', () => {
+  const moduleMocker = new ModuleMocker(global);
   let app: INestApplication;
-  let userModule: UserModule;
+  let user: User;
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const moduleFixture: TestingModule = await Test.createTestingModule({
       imports: [UserModule],
     }).compile();
     app = moduleFixture.createNestApplication();
     await app.init();
   });
 
-  //TODO: IT Block
-  //it('/ (GET) ')
+  it('/GET ', () => {
+    return request(app.getHttpServer()).get('/user').expect(200).expect('[]');
+  });
+
+  it('/user/signUp ', () => {
+    return request(app.getHttpServer())
+      .post('/user/signUp')
+      .field(
+        'user',
+        'username: "john", password: "john", mail: "john@email.com"',
+      )
+      .attach('', '');
+  });
+
+  it('/update', () => {
+    return request(app.getHttpServer())
+      .get('/user/update')
+      .expect(200)
+      .expect('');
+  });
+
+  it('delete', () => {
+    return request(app.getHttpServer())
+      .get('/user/delete')
+      .expect(200)
+      .expect('');
+  });
```

Comment:
What are you updating/deleting here?
You give it no `:id` or body - how would it know what to delete/update?

Both (update & delete) routes require an id, as you see in the controller!

And another thing: the order in which the tests are executed may be undefined! What I'm implying is that you cannot assume when you do the 'delete' test, that the `/user/signUp` test has run already!
Which is why you need to create and delete the user in each test you create from the beginning!

That's also pretty much the only reasonable way to get the id of the user you want to delete!
Meaning: you first request the user and expect it to return an ID, which you then can use to delete the user later on (once you finished the tests!)

That also reduces the amount of test-cases to 2:
- `it('User Signup', () => {});`
- `it('User Update', () => {});`
Since deletion and creating will be tested in all of them either way!